src/core/fieldformatter/qgsvaluerelationfieldformatter.h

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+  qgsvaluerelationfieldformatter.h - QgsValueRelationFieldFormatter
+
+ ---------------------
+ begin                : 3.12.2016
+ copyright            : (C) 2016 by Matthias Kuhn
+ email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSVALUERELATIONFIELDKIT_H
+#define QGSVALUERELATIONFIELDKIT_H
+
+#include "qgsfieldformatter.h"
+
+#include <QVector>
+#include <QVariant>
+
+class CORE_EXPORT QgsValueRelationFieldFormatter : public QgsFieldFormatter
+{
+  public:
+    struct ValueRelationItem
```

Comment:
It was before.
I changed it exactly because `first` and `second` are not verbose at all. (See also [this post](https://www.kdab.com/tuple-pair-cpp-apis/))